### ISSUE TITLE
Stable branch policy

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-terra#2352]
# What is the expected enhancement?

Right now we have a stable branch for making bug fix releases, but without a policy for what and how we include changes in this. To help facilitate using it moving forward I wanted to propose a policy for this. I wrote this based on similar policies I've used on other open source projects:

## Stable Policy

The stable branch is intended to be a safe source of fixes for high impact bugs and security issues which have been fixed on master since a release. When reviewing a stable branch PR we need to balance the risk of any given patch with the value that it will provide to users of the stable branch. Only a limited class of changes are appropriate for inclusion on the stable branch. A large, risky patch for a major issue might make sense. As might a trivial fix for a fairly obscure error handling case. A number of factors must be weighed when considering a change:

- The risk of regression: even the tiniest changes carry some risk of breaking something and we really want to avoid regressions on the stable branch
- The user visible benefit: are we fixing something that users might actually notice and, if so, how important is it?
- How self-contained the fix is: if it fixes a significant issue but also refactors a lot of code, it’s probably worth thinking about what a less risky fix might look like
- Whether the fix is already on master: a change must be a backport of a change already merged onto master, unless the change simply does not make sense on master.

## Backporting procedure:

When backporting a patch from master to stable we want to keep a reference to the change on master. When you create the branch for the stable PR you can use:

`$ git cherry-pick -x $master_commit_id`

If the patch you’re proposing will not cherry-pick cleanly, you can help by resolving the conflicts yourself and proposing the resulting patch. Please keep Conflicts lines in the commit message to help review of the stable patch.

## Backport Tags

Bugs or PRs tagged with `stable backport potential` are bugs which apply to the stable release too and may be suitable for backporting once a fix lands in master. Once the backport has been proposed, the tag should be removed.

**Comment by taalexander** [comment]
Perhaps the tag should be `stable backport potential` as existing tags use spaces.

**Comment by mtreinish** [comment]
Sure, spaces are fine. I'm used to a more cli driven world and forget that github supports spaces. I've updated it in the issue.

**Comment by kdk** [comment]
:+1: on this idea. Question on the backporting procedure, what is the advantage to fixing in master first, and then cherry-picking to stable? The alternative would be to fix first in stable, and then merge into master. The only difference I can see is in where conflicts have to be resolved (when merging into master, or when merging in to stable.)

**Comment by mtreinish** [comment]
It's mostly about managing risk of regression by introducing a new change on stable. By going through master first we're already reviewing and testing it to merge on master before we introduce the change to stable. The idea is that we only want to make changes to stable when the change is a known quantity. There will be merge conflicts in either case it doesn't really matter which one we start on when cherry-picking between branches since it'll be roughly the same amount of work.
